### PR TITLE
do not truncate multiline key/value in view

### PR DESCRIPTION
**Pull request by mantis-toboggan-md**
Per @deniseschannon request, this PR un-hides the cert data in the secret detail view of service acct tokens (and other secrets with long data values). `KeyValue` view-mode styling truncates all long `key` and `value` cells, but I think it makes sense for multiline inputs to remain multiline in view mode.
Screen Shot 2020-06-25 at 3 34 20 PM

### CODE REVIEW

**Comment by lvuch** [comment]
I agree it should wrap but I think @vincent99 may have specific feedback on implementation.

**Comment by vincent99** [comment]
First, the value should have the CR/LFs converted to `<br>` so that line breaks in the actual value are displayed properly (or you could use `<pre>`, but keep reading).   In the pictured example the `----- END CERTIFICATE -----` at the end belongs on its own line.  This only sorta kinda looks like it's working now because the cert is broken up into lines of about the same width as the available width to display it.

Then you decide what to do with lines that are still too wide for the space available:
  - Clip
  - Scroll
  - Force wrapping (break-word)
 
I argue that the "detail" page for a resource should never clip the value.  The point of the detail is to show me the value and there's no other place to get the full unclipped value (except going to edit, which I may not have permission to do).  So I would go with force wrapping (`.force-wrap` class).

Finally, there's not much reason for the Key to be 50% of the width.  You can pick a smaller width, or possibly make the table not `table-layout: fixed` for this case.  @lvuch was working on stuff in this area though so coordinate with her.

(Also your `hide-overflow` class is effectively the same as the already existing global `styles/base/_mixins` -> `clip`)

File: components/form/KeyValue.vue [hunk 1]
```diff
@@ -348,10 +355,33 @@ export default {
...
+        this.$emit('input', out);
       }
-      this.$emit('input', out);
-    }
-  },
+    },
+
+    displayProps(value) {
+      const binary = typeof value === 'string' && !asciiLike(value);
+      const isFormatted = typeof value === 'string' ? value.includes('-----BEGIN') : false;
```
Comment (vincent99) [inline components/form/KeyValue.vue:364]: Looking for `-----BEGIN` is not a good assumption... lots of other kinds of values (YAML, JSON, TOML, etc) have newlines in them and they should be shown with them.

Using `<pre>` also gets you back into horizontal scrolling.

```javascript
displayValue = escapeHtml(value).replace(/(\r\n|\r|\n)/g, '<br/>\n');
```

```html
<span v-html="row.displayValue"/>
```

**Comment by mantis-toboggan-md** [comment]
Updates:
* detail view of KeyValue never clips value
* newline characters converted to <br/>
* formatting and syntax highlighting for JSON (I went with base16 color theme since that seems to be what CodeMirror uses)
* values over 10kB have 'n Bytes [Click to Show]'

I think this could also include syntax highlighting for a lot of other file types that show up potentially. I can do that if this looks good:
Screen Shot 2020-06-29 at 3 04 58 PM

There's already a ticket for converting KeyValue (and other components) from table to grid so I'm not going to mess with that there.

**Comment by vincent99** [comment]
Looks nice;  can you render it with a codemirror-based component (no line numbers or gutters, line wrapping, not editable, ...) instead of loading another highlight lib/color scheme?

Key should top-aligned, for a long value the key is hard to find in the middle (this can go in the other PR if it makes more sense there).

**Comment by mantis-toboggan-md** [comment]
Easy enough to switch to using CodeMirror, but I am not a fan of the background color, in or out of tabs:
Screen Shot 2020-06-30 at 11 02 59 AM

**Comment by vincent99** [comment]
Indeed that's not great.. is it somewhere we can easily add an option to override the theme's background color instead of defining a whole separate theme with a slightly different background?  Or change the theme to use transparent and put the dark color behind/outside of codemirror where needed.

File: components/YamlEditor.vue [hunk 2]
```diff
@@ -200,6 +200,19 @@ export default {
...
+  ::v-deep #code-mirror  {
+      position: relative;
+      .CodeMirror {
+        height: initial;
+        position: absolute;
+        top:0;
+        bottom:0;
+        left:0;
+        right:0;
+        background-color: #151515;
```
Comment (vincent99) [inline components/YamlEditor.vue:212]: Add a variable to theme and appropriate light color value

**Pull request merged**